CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/CameraFragment.kt

```diff
@@ -187,37 +183,38 @@ class CameraFragment : Fragment() {
     }
 
     /** Define callback that will be triggered after a photo has been taken and saved to disk */
-    private val imageSavedListener = object : ImageCapture.OnImageSavedListener {
-        override fun onError(
-                error: ImageCapture.ImageCaptureError, message: String, exc: Throwable?) {
-            Log.e(TAG, "Photo capture failed: $message")
-            exc?.printStackTrace()
+    private val imageSavedListener = object : ImageCapture.OnImageSavedCallback {
+        override fun onError(imageCaptureError: Int, message: String, cause: Throwable?) {
+            Log.e(TAG, "Photo capture failed: $message", cause)
         }
 
         override fun onImageSaved(photoFile: File) {
             Log.d(TAG, "Photo capture succeeded: ${photoFile.absolutePath}")
 
             // We can only change the foreground Drawable using API level 23+ API
             if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
-
                 // Update the gallery thumbnail with latest picture taken
                 setGalleryThumbnail(photoFile)
             }
 
-            // Implicit broadcasts will be ignored for devices running API
-            // level >= 24, so if you only target 24+ you can remove this statement
+            // Implicit broadcasts will be ignored for devices running API level >= 24
+            // so if you only target API level 24+ you can remove this statement
             if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
                 requireActivity().sendBroadcast(
-                        Intent(Camera.ACTION_NEW_PICTURE, Uri.fromFile(photoFile)))
+                        Intent(android.hardware.Camera.ACTION_NEW_PICTURE, Uri.fromFile(photoFile))
+                )
             }
 
             // If the folder selected is an external media directory, this is unnecessary
             // but otherwise other apps will not be able to access our images unless we
             // scan them using [MediaScannerConnection]
-            val mimeType = MimeTypeMap.getSingleton()
-                    .getMimeTypeFromExtension(photoFile.extension)
-            MediaScannerConnection.scanFile(
-                    context, arrayOf(photoFile.absolutePath), arrayOf(mimeType), null)
+            val mimeType = MimeTypeMap.getSingleton().getMimeTypeFromExtension(photoFile.extension)
+            // It prevents a force close, when pressing the photo-view button early on.
+            if(!requireActivity().isFinishing) {
```

Comment:
I'm still unable to reproduce this issue. You mentioned that this gets triggered by pressing the shutter button, but that alone should not be finishing the activity. I would remove this check and, if you are able to provide steps to reliably reproduce this, we can discuss a fix.

Reply:
Not the shutter button, that photo-view / gallery button ...probably that "service connection lost" only may happen, when there are a few more images to scan, so that it would take more time to scan them. I noticed this when fixing the other other on `IllegalArgumentException`; my fault that I didn't take a note on how to crash it.

Reply:
I also tried pressing the gallery button as quickly as possible upon startup, and I still can't reproduce. I suggest tracking this as a separate issue.

Reply:
It's `onImageSaved()`, so it's a) first the shutter button and b) then the photo-view button.

Reply:
I still can't reproduce. I am seeing an error for the photo capture failing, because the camera closes before the capture completes: `androidx.camera.core.CameraClosedException: Camera is closed.`

So we can potentially disable the gallery button until capture finishes. Either way, I suggest tracking this as a separate issue.

Reply:
Have removed it, if ever the error message will be "service connection lost", you'll know what it is.